packages/harfbuzz.rb

```diff
@@ -64,4 +67,39 @@ def self.build
   def self.install
     system "DESTDIR=#{CREW_DEST_DIR} ninja install -C builddir"
   end
+
+  def self.preinstall
+    @device = JSON.parse(File.read("#{CREW_CONFIG_PATH}device.json"), symbolize_names: true)
+    if @device[:installed_packages].any? { |elem| elem[:name] == 'freetype' }
+      system "sed -i '/freetype2/d;/libfreetype/d' filelist"
+      system "sed -i '/freetype2/d;/libfreetype/d' dlist"
+    end
+  end
+
+  def self.postinstall
+    # This should become a function.
+    # check for conflicts with other installed files
+    @override_allowed = %w[fontconfig cairo]
+    puts "Checking for conflicts with files from installed packages..."
+    conflicts = []
+    conflictscmd = %x[grep --exclude #{CREW_META_PATH}#{self.name}.filelist -Fxf #{CREW_META_PATH}#{self.name}.filelist #{CREW_META_PATH}*.filelist]
+    conflicts << conflictscmd.gsub(/(\.filelist|#{CREW_META_PATH})/, '').split("\n")
+    conflicts.reject!(&:empty?)
+    unless conflicts.empty?
+      if self.conflicts_ok?
+        puts "Warning: There is a conflict with the same file in another package.".orange
+      else
+        puts "Error: There is a conflict with the same file in another package.".lightred
+        @_errors = 1
+      end
+      conflicts.each do |conflict|
+        conflict.each do |thisconflict|
+          singleconflict = thisconflict.split(':',-1)
+          if @override_allowed.include?(singleconflict[0])
+            system "sed -i '\\\?^#{singleconflict[1]}?d'  #{CREW_META_PATH}/#{singleconflict[0]}.filelist"
+          end
+        end
+      end
+    end
+  end
```

Comment:
Should this logic replace the conflict handling code we have in crew? Seems like it could be used universally?